pkg/operator/ceph/csi/csi.go

```diff
@@ -40,9 +44,13 @@ func ValidateAndConfigureDrivers(context *clusterd.Context, namespace, rookImage
 	}
 
 	if CSIEnabled() {
-		if err := startDrivers(context.Clientset, context.RookClientset, namespace, serverVersion, ownerInfo, v); err != nil {
-			logger.Errorf("failed to start Ceph csi drivers. %v", err)
-			return
+		maxRetries := 3
+		for i := 0; i < maxRetries; i++ {
+			if err = startDrivers(context.Clientset, context.RookClientset, namespace, serverVersion, ownerInfo, v); err != nil {
+				logger.Errorf("failed to start Ceph csi drivers, will retry starting csi drivers %d more times. %v", maxRetries-i-1, err)
```

Comment:
I always hesitate to suggest a sleep, but I think we should add a short one between retries in case there was some issue that needs a few seconds to settle.
```suggestion
				logger.Errorf("failed to start Ceph csi drivers, will retry starting csi drivers %d more times. %v", maxRetries-i-1, err)
				// add a sleep for 5s
```

Reply:
During my initial review, I was also ok not having a sleep. Honestly, it's so difficult to tell whether 5sec is going to change anything, 5sec feels really long. REST calls are slow by nature so I'd prefer not to add a sleep in this case.

Reply:
I guess I'm ok either way, it's just a small chance it would make a difference. We can remove it.

Reply:
Removed sleep, PTAL @travisn @leseb